### PR TITLE
dashboards: Add Yaxis.AutoMin, YAxis.AutoMax, and custom Unmarshal

### PR DESCRIPTION
**Pull request by anthony-tsang**
* Add in the AutoMin and AutoMax fields for the Yaxis struct so we can work around the times when datadog's API returns a string when we expect a float64.
* Add a custom UnmarshalJSON for Yaxis to utilize the new AutoMin and AutoMax fields.

related error: "json: cannot unmarshal string into Go struct field Yaxis.min of type
float64"

Followed the suggestions from @yfronto in the [previous open PR](https://github.com/zorkian/go-datadog-api/pull/147).

Please let me know if this is the right direction and if you have anymore suggestions.

### CODE REVIEW

**Comment by jamesmharvey** [comment]
This is in reference to issue #103

**Comment by yfronto** [comment]
Thanks for the submission. Generally looks good. Can you run `make generate`?

File: dashboards.go [hunk 1]
```diff
@@ -53,10 +54,52 @@ type GraphEvent struct {
...
+		Max *string `json:"max,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(y),
+	}
+	if err := json.Unmarshal(data, wrapper); err != nil {
+		return err
+	}
+
+	if *wrapper.Min == "auto" {
```
[the remark this answers is not in the document]
Comment (anthony-tsang) [reply dashboards.go]: Added the additional nil check

File: dashboards.go [hunk 2]
```diff
@@ -53,10 +54,52 @@ type GraphEvent struct {
...
+		*y.AutoMin = true
```
[the remark this answers is not in the document]
Comment (anthony-tsang) [reply dashboards.go]: removed pointers for the Auto* fields

File: dashboards.go [hunk 3]
```diff
@@ -53,10 +54,52 @@ type GraphEvent struct {
...
+		y.Min = nil
+	} else {
+		*wrapper.AutoMin = false
```
[the remark this answers is not in the document]
Comment (anthony-tsang) [reply dashboards.go]: Just for clarification are you saying I do not need to set wrapper.AutoMin or the whole else block is not necessary?

File: dashboards.go [hunk 4]
```diff
@@ -53,45 +52,46 @@ type GraphEvent struct {
 }
 
 type Yaxis struct {
-	Min   *float64 `json:"min,omitempty"`
-	AutoMin *bool
-	Max   *float64 `json:"max,omitempty"`
-	AutoMax * bool
-	Scale *string  `json:"scale,omitempty"`
+	Min     *float64 `json:"min,omitempty"`
+	AutoMin bool     `json:"-"`
```
Comment (yfronto) [inline dashboards.go]: Also FYI, making a field private will also prevent it from being marshalled (although it also prevents consumers from using it, obviously).

**Pull request merged**